swapi/src/app/sign-up/sign-up.component.html

```diff
@@ -0,0 +1,79 @@
+<div class="container">
+  <form [formGroup]="signUpForm" (ngSubmit)="signUp()">
+    <mat-form-field>
+      <mat-label>Email:</mat-label>
+      <input
+        matInput
+        type="text"
+        inputmode="email"
+        formControlName="email"
+        placeholder="example@abc.com">
+      <mat-icon matSuffix>mail_outline</mat-icon>
+      <mat-error *ngIf="signUpForm.controls.email.errors?.['required']">
+        Email is required
+      </mat-error>
+      <mat-error *ngIf="signUpForm.controls.email.errors?.['email']">
+        Please enter valid email
+      </mat-error>
+    </mat-form-field>
+
+    <mat-form-field>
+      <mat-label>Password:</mat-label>
+      <input
+        matInput
+        [type]="hideEmail ? 'password' : 'text'"
+        formControlName="password"
+        placeholder="At least 6 characters">
+      <button
+        type="button"
+        mat-icon-button
+        matSuffix
+        (click)="hideEmail = !hideEmail"
+        [attr.aria-label]="'Hide password'"
+        [attr.aria-pressed]="hideEmail">
+        <mat-icon>{{ hideEmail ? 'visibility_off' : 'visibility' }}</mat-icon>
+      </button>
+      <mat-error
+        *ngIf="signUpForm.controls.password.errors?.['required']">
+        Password is required
+      </mat-error>
+      <mat-error
+        *ngIf="signUpForm.controls.password.errors?.['minlength']">
+        Password should contain at least 6 characters
+      </mat-error>
+    </mat-form-field>
+
+    <mat-form-field>
+      <mat-label>Confirm password:</mat-label>
+      <input
+        matInput
+        [type]="hideConfirmEmail ? 'password' : 'text'"
+        formControlName="confirmPassword">
+      <button
+        type="button"
+        mat-icon-button
+        matSuffix
+        (click)="hideConfirmEmail = !hideConfirmEmail"
+        [attr.aria-label]="'Hide password'"
+        [attr.aria-pressed]="hideConfirmEmail">
```

Comment:
cool stuff

Reply:
is it really good approach? please resolve in this case